core/logging.py

```diff
@@ -52,27 +46,98 @@ def emit(self, record: logging.LogRecord):
+        # truncate the pathname
+        if "/deploy" in record.pathname:
+            pathname = record.pathname.split("/deploy/")[1]
+        else:
+            # When we have a module in a very deep package path, like
+            # google.cloud.resourcemanager_v3.services.projects.client,
+            # we turn it into something like google/.../projects/client.
+            pathname = record.pathname
+            if len(pathname) > 20:
+                parts = pathname.split("/")
+                del parts[1:-3]
+                parts.insert(1, "...")
+                pathname = "/".join(parts)
+
+        record.pathname = pathname
+
+        msg = super().format(record)
+
+        # colorize entire message according to the levelname
+        if record.levelname in ViURLocalFormatter.LEVELS:
+            colorize = ViURLocalFormatter.COLOR_SEQ % (ViURLocalFormatter.LEVELS[record.levelname] + 30)
+            return colorize + msg + ViURLocalFormatter.RESET_SEQ
+
+        return msg
+
+
+# Logger config
+
+client = google.cloud.logging.Client()
+requestLogger = client.logger("ViUR")
+requestLoggingRessource = Resource(
+    type="gae_app",
+    labels={
+       "project_id": conf["viur.instance.project_id"],
+       "module_id": "default",
+       "version_id": conf["viur.instance.app_version"] if not conf[
+           "viur.instance.is_dev_server"] else "dev_appserver",
+    }
+)
 
 logger = logging.getLogger()
 logger.setLevel(logging.DEBUG)
 
-for loggerName, level in oldLevels.items():
-    logging.getLogger(loggerName).setLevel(level)
+# Inherit old logger levels
+for name, level in {
+            k: v.getEffectiveLevel()
+            for k, v in logging.root.manager.loggerDict.items()
+            if isinstance(v, logging.Logger)
+        }.items():
+    logging.getLogger(name).setLevel(level)
 
+# Remove handlers from logger
 for handler in logger.handlers[:]:
     logger.removeHandler(handler)
 
-handler = ViURDefaultLogger(client, name="ViUR-Messages", resource=Resource(type="gae_app", labels={}))
-logger.addHandler(handler)
+if not conf["viur.instance.is_dev_server"]:
+    # Plug-in ViURDefaultLogger
+    handler = ViURDefaultLogger(client, name="ViUR-Messages", resource=Resource(type="gae_app", labels={}))
+    logger.addHandler(handler)
 
-sh = logging.StreamHandler()
-formatter = logging.Formatter("%(levelname)-8s %(asctime)s %(filename)s:%(lineno)s] %(message)s")
-sh.setFormatter(formatter)
-logger.addHandler(sh)
+    # Custom formatter when running in the cloud
+    formatter = logging.Formatter("%(levelname)-8s %(asctime)s %(filename)s:%(lineno)s] %(message)s")
+
+else:
+    # Use ViURLocalFormatter for local debug message formatting
+    formatter = ViURLocalFormatter(f"[%(asctime)s] %(pathname)s:%(lineno)d [%(levelname)s] %(message)s")
+    sh = logging.StreamHandler()
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
 
+# https://github.com/googleapis/python-logging/issues/13#issuecomment-539723753
```

Comment:
Since `sh` is only defined in the `else:` block, we need to move this as well.

Reply:
fixed in 4ff488d